lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -148,45 +147,62 @@ const AbstractDialog = forwardRef<HTMLDialogElement, Props>((props, ref) => {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       const dialogEl = dialogRef.current!;
 
-      // When tab key is pressed
-      // cycles focus on first or last focusable item (if exists)
-      // Do the same if focus is not IN the dialog to ensure focus is trapped within.
-      if (event.key === 'Tab') {
-        const firstFocusableElement = getFirstVisibleFocusableElement(dialogEl);
-        const lastFocusableElement = getLastVisibleFocusableElement(dialogEl);
-        const activeFocusIsNullOrDialog
-          = [document.body, dialogEl, null].includes(document.activeElement as HTMLElement);
-        const activeFocusIsInDialog = !activeFocusIsNullOrDialog && dialogEl.contains(document.activeElement);
-
-        if (!activeFocusIsInDialog) {
-          event.preventDefault();
-        }
-
-        if (event.shiftKey) {
-          if (
-            (!activeFocusIsInDialog || document.activeElement === firstFocusableElement)
-            && lastFocusableElement
-          ) {
-            lastFocusableElement.focus();
-            event.preventDefault();
+      // Only manage tab cycling when the focus is within the dialog
+      // and not nested inside another dialog
+      // so that there is no keynav conflict when another modal is open.
+      if (
+        document.activeElement
+        && document.activeElement.closest(DIALOG_MODAL_SELECTOR) === dialogEl
+      ) {
+        // Cycle focus on the first or last focusable item (if exists).
+        // Do the same if focus is NOT in the dialog to ensure that the focus is trapped within.
+        if (event.key === 'Tab') {
+          const firstFocusableElement = getFirstVisibleFocusableElement(dialogEl);
+          const lastFocusableElement = getLastVisibleFocusableElement(dialogEl);
+
+          if (event.shiftKey) {
+            if (document.activeElement === firstFocusableElement) {
+              lastFocusableElement.focus();
+              event.preventDefault();
+            }
           }
-        }
-        else {
-          if (
-            (!activeFocusIsInDialog || document.activeElement === lastFocusableElement)
-            && firstFocusableElement
-          ) {
+          else if (document.activeElement === lastFocusableElement) {
             firstFocusableElement.focus();
             event.preventDefault();
           }
         }
       }
     };
 
+    // Prevent leak when focus goes out of the dialog modal.
+    const handleFocusOut = (event: FocusEvent) => {
+      const dialogEl = dialogRef.current;
+      if (dialogEl) {
+        const departingFocusIsInsideDialog = !!(
+          event.target
+          && (event.target as HTMLElement).closest(DIALOG_MODAL_SELECTOR) === dialogEl
+        );
+        const receivingFocusIsInsideDialog
+          = !!(
+            event.relatedTarget
+            && (event.relatedTarget as HTMLElement).closest(DIALOG_MODAL_SELECTOR)
+          );
```

Comment:
These two statements follow a very similar pattern but are formatted fairly differently.  I prefer the formatting on the first one.

Reply:
> These two statements follow a very similar pattern but are formatted fairly differently. I prefer the formatting on the first one.

Fixed: 8eb61a1a01d8388378b93f387f2ffb84dee6057b